CMSgov/design-system
# [NO TICKET] Update dist folder structure

This is a follow up to [this PR](https://github.com/CMSgov/design-system/pull/707)

### Summary
- Dist folders structure is updated to this:
```
└── @cmsgov/design-system/dist
    ├── components/
    │   └── index.js        Compiled JS (CommmonJS)
    ├── css/
    │   └── index.css       Compiled CSS
    ├── esnext/
    │   └── index.esm.js    Compiled JS (ES Module)
    ├── fonts/
    ├── images/
    └── scss/
```
- src folders structure is updated to this:
```
└── @cmsgov/design-system/src
    ├── components/
    │   └── index.js        Uncompiled JS
    ├── fonts/
    ├── images/
    └── styles/
        └── index.scss       Uncompiled SCSS
```
- Imports are updated
- Documentation is updated

### How to test
- Run `yarn start` and ensure the `dist` folder is working correctly
- Review all the changed documentation
- Test the build scripts with the child design system
